File: Blog.DAL/Models/Article.cs
namespace Blog.DAL.Models;

public class Article : Commentable
{
    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
    public string Title { get; set; }
    public string Description { get; set; }
    public string Content { get; set; }
    
    public User Author { get; set; }
    public Rubric Rubric { get; set; }
}

File: Blog.ClientApp/src/app/models/article.ts
import { Profile } from "./profile";

export interface Article {
    id: string;
    createdDate: Date | null;
    title: string;
    description: string;
    content: string;
    isAuthor: boolean;
    author?: Profile;
    rubricName: string;
    rubricId: string;
}

export class Article implements Article {
    constructor(init?: ArticleFormValues) {
        Object.assign(this, init);
    }
}

export class ArticleFormValues {
    id?: string = undefined;
    title: string = '';
    description: string = '';
    content: string = '';
    createdDate: Date | null = new Date();
    rubricName: string = '';
    rubricId: string = '';

    constructor(article?: ArticleFormValues) {
        if (article) {
            this.id = article.id;
            this.title = article.title;
            this.description = article.description;
            this.content = article.content;
            this.rubricName = article.rubricName;
            this.rubricId = article.rubricId;
        }
    }
}